### PR TITLE
M2-249: Applet table - Transfer ownership popup

### PR DESCRIPTION
**Pull request by nmialik**

### CODE REVIEW

File: src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    const { transferOwnership } = folders.thunk;
+    const result = await dispatch(transferOwnership({ appletId: item.id, email }));
+
+    if (transferOwnership.fulfilled.match(result)) {
+      setTransferOwnershipPopupVisible(false);
+    } else if (transferOwnership.rejected.match(result)) {
+      const errorObj = result.payload as AxiosError;
+      const errorData = errorObj.response?.data as AxiosError<ErrorResponse>;
+      if (errorData) {
+        setErrorMessage(errorData.message);
```
Comment (kdanilenka32) [inline src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx:48]: you can do setErrorMessage(errorData.message || errorObj.message)

File: src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        email: yup.string().email(t('incorrectEmail')!),
+      }),
+    ),
+    defaultValues: { email: '' },
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleTransferOwnership = async ({ email }: TransferOwnership) => {
+    const { transferOwnership } = folders.thunk;
+    const result = await dispatch(transferOwnership({ appletId: item.id, email }));
```
Comment (kdanilenka32) [inline src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx:40]: you can use just hook useAsync

File: src/components/Popups/DeletePopup/DeletePopup.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { useTranslation } from 'react-i18next';
+
```
Comment (kdanilenka32) [inline src/components/Popups/DeletePopup/DeletePopup.tsx:2]: My suggestion keep in popup global components

File: src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx (hunk 1 above)
Comment (kdanilenka32) [reply src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx:48]: if you use useAsync hook you no need the error state

File: src/components/Tables/AppletsTable/Table/AppletItem/AppletItem.tsx [hunk 4]
```diff
@@ -69,11 +73,20 @@ export const AppletItem = ({ item }: { item: FolderApplet }) => {
           <Actions items={actionsRender(actions)} context={item} />
         </TableCell>
       </TableRow>
-      <DeletePopUp
-        deleteModalVisible={deleteModalVisible}
-        setDeleteModalVisible={setDeleteModalVisible}
-        item={item}
-      />
+      {deletePopupVisible && (
```
Comment (kdanilenka32) [inline src/components/Tables/AppletsTable/Table/AppletItem/AppletItem.tsx:76]: Why here  {deletePopupVisible && ( statement

File: src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    if (transferOwnership.fulfilled.match(result)) {
+      setTransferOwnershipPopupVisible(false);
```
Comment (kdanilenka32) [inline src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.tsx:43]: pls create transferOwnershipPopupClose func

File: src/components/Tables/AppletsTable/Table/AppletItem/AppletItem.tsx (hunk 4 above)
Comment (nmialik) [reply src/components/Tables/AppletsTable/Table/AppletItem/AppletItem.tsx:76]: why do we need to store the component and its state after closing?

File: src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.styles.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Box, styled } from '@mui/material';
+import { StyledLabelMedium, StyledTitleMedium } from 'styles/styledComponents/Typography';
+
```
Comment (kdanilenka32) [inline src/components/Popups/TransferOwnershipPopup/TransferOwnershipPopup.styles.ts:3]: pls check spaces here

File: src/components/Tables/AppletsTable/Table/AppletItem/AppletItem.tsx (hunk 4 above)
Comment (kdanilenka32) [reply src/components/Tables/AppletsTable/Table/AppletItem/AppletItem.tsx:76]: ok, make sense

File: src/redux/modules/Folders/Folders.reducer.ts [hunk 7]
```diff
@@ -70,65 +74,77 @@ export const extraReducers = (builder: ActionReducerMapBuilder<FoldersSchema>):
     }
   });
 
-  builder.addCase(saveFolder.pending, ({ foldersApplets }, action) => {
-    createFoldersPendingData(foldersApplets, action.meta.requestId);
+  // // // // // // // // // // // // // // // // // // // // // // // // // // // //
```
Comment (kdanilenka32) [inline src/redux/modules/Folders/Folders.reducer.ts:77]: удоли))

Comment (nmialik) [reply src/redux/modules/Folders/Folders.reducer.ts:77]: ой))

File: src/components/Tables/AppletsTable/Table/FolderItem/FolderItem.tsx [hunk 8]
```diff
@@ -38,23 +40,44 @@ export const FolderItem = ({ item }: FolderItemProps) => {
...
+  };
+
+  const saveFolder = () => {
+    if (!folder.isNew && folder.name === item.name) {
+      return setFolder((folder) => ({ ...folder, isRenaming: false }));
+    }
+    const isNameExist = foldersApplets.find(
+      (folderApplet) => folderApplet.isFolder && folderApplet.name === folder.name?.trim(),
+    );
+    const name = !isNameExist && folder.name?.trim() ? folder.name?.trim() : item.name;
```
Comment (kdanilenka32) [inline src/components/Tables/AppletsTable/Table/FolderItem/FolderItem.tsx:73]: (!isNameExist && folder.name?.trim()) || item.name; 
if this way?

File: src/components/Tables/AppletsTable/Table/FolderItem/FolderItem.tsx [hunk 9]
```diff
@@ -74,12 +97,23 @@ export const FolderItem = ({ item }: FolderItemProps) => {
...
                 endAdornment={
                   <InputAdornment position="end">
-                    <Svg id="cancel-rounded" />
+                    <StyledCloseButton
+                      onMouseDown={(e) => {
+                        e.preventDefault();
+                        handleClearClick();
+                      }}
+                    >
+                      <Svg width={24} height={24} id="cross" />
```
Comment (kdanilenka32) [inline src/components/Tables/AppletsTable/Table/FolderItem/FolderItem.tsx:115]: size 24 by default

File: src/redux/modules/Folders/Folders.thunk.ts [hunk 10]
```diff
@@ -185,3 +186,14 @@ export const removeAppletFromFolder = createAsyncThunk(
     }
   },
 );
+
+export const transferOwnership = createAsyncThunk(
```
Comment (kdanilenka32) [inline src/redux/modules/Folders/Folders.thunk.ts:190]: Are you using this?

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**